src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
-                    //Trying to serve untilour last breath
-                    ResendAll();
-                    _writer.Tell(EndpointWriter.FlushAndStop.Instance);
-                    Context.Become(FlushWait);
-                })
-                .With<EndpointManager.Send>(HandleSend)
-                .With<Ack>(ack =>
+            Receive<EndpointWriter.FlushAndStop>(flush =>
+            {
+                //Trying to serve until our last breath
+                ResendAll();
+                _writer.Tell(EndpointWriter.FlushAndStop.Instance);
+                Become(FlushWait);
+            });
+            Receive<IsIdle>(idle => { }); // Do not reply, we will Terminate soon, or send a GotUid
+            Receive<EndpointManager.Send>(send => HandleSend(send));
+            Receive<Ack>(ack =>
+            {
+                // If we are not sure about the UID just ignore the ack. Ignoring is fine.
+                if (UidConfirmed)
                 {
-                    if (!UidConfirmed) _pendingAcks.Add(ack);
-                    else
+                    try
                     {
-                        try
-                        {
-                            _resendBuffer = _resendBuffer.Acknowledge(ack);
-                        }
-                        catch (Exception ex)
-                        {
-                            throw new InvalidAssociationException(
-                                string.Format(
-                                    "Error encountered while processing system message acknowledgement {0} {1}",
-                                    _resendBuffer, ack), ex);
-                        }
+                        _resendBuffer = _resendBuffer.Acknowledge(ack);
+                    }
+                    catch (Exception ex)
+                    {
+                        throw new HopelessAssociation(_localAddress, _remoteAddress, Uid,
+                            new IllegalStateException($"Error encountered while processing system message acknowledgement buffer: {_resendBuffer} ack: {ack}", ex));
+                    }
 
-                        if (_lastCumulativeAck < ack.CumulativeAck)
-                        {
-                            _lastCumulativeAck = ack.CumulativeAck;
-                            // Cumulative ack is progressing, we might not need to resend non-acked messages yet.
-                            // If this progression stops, the timer will eventually kick in, since scheduleAutoResend
-                            // does not cancel existing timers (see the "else" case).
-                            RescheduleAutoResend();
-                        }
-                        else
-                        {
-                            ScheduleAutoResend();
-                        }
+                    ResendNacked();
+                }
+            });
+            Receive<AttemptSysMsgRedelivery>(sysmsg =>
+            {
+                if (UidConfirmed) ResendAll();
```

Comment:
Message handling for the normal system message redelivery timer. This will only run if the UID is confirmed and will only do work if there are messages that actually need to be resent.